cmd/standalone/main.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"flag"
+	"net"
+	"net/http"
+	"runtime"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/uber-go/zap"
+	"github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/thrift"
+
+	"github.com/uber/jaeger-lib/metrics"
+	"github.com/uber/jaeger-lib/metrics/go-kit"
+	"github.com/uber/jaeger-lib/metrics/go-kit/expvar"
+	agentApp "github.com/uber/jaeger/cmd/agent/app"
+	basicB "github.com/uber/jaeger/cmd/builder"
+	collectorB "github.com/uber/jaeger/cmd/collector/app/builder"
+	queryApp "github.com/uber/jaeger/cmd/query/app"
+	queryB "github.com/uber/jaeger/cmd/query/app/builder"
+	"github.com/uber/jaeger/pkg/recoveryhandler"
+	"github.com/uber/jaeger/storage/spanstore/memory"
+	jc "github.com/uber/jaeger/thrift-gen/jaeger"
+	zc "github.com/uber/jaeger/thrift-gen/zipkincore"
+)
+
+// standalone/main is a standalone full-stack jaeger backend, backed by a memory store
+func main() {
+	logger := zap.New(zap.NewJSONEncoder())
+	metricsFactory := xkit.Wrap("jaeger-standalone", expvar.NewFactory(10))
+	memStore := memory.NewStore()
+	startAgent(logger, metricsFactory)
+	startCollector(logger, metricsFactory, memStore)
+	startQuery(logger, metricsFactory, memStore)
+
+	select {}
+}
+
+func startAgent(logger zap.Logger, baseFactory metrics.Factory) {
+	builder := agentApp.NewBuilder()
+	builder.Bind(flag.CommandLine)
+	flag.Parse()
```

Comment:
parsing  flags should be in main, and you need to bind all flags from all components